tests/cpp/test_host_ir_llvm_lowering.cpp

```diff
@@ -0,0 +1,269 @@
+// clang-format off
+/*
+* SPDX-FileCopyrightText: Copyright (c) 2025-present NVIDIA CORPORATION & AFFILIATES.
+* All rights reserved.
+* SPDX-License-Identifier: BSD-3-Clause
+*/
+// clang-format on
+#include <gmock/gmock-more-matchers.h>
+
+#include <fusion.h>
+#include <global_allocator.h>
+#include <host_ir/container.h>
+#include <host_ir/executor.h>
+#include <ir/all_nodes.h>
+#include <ops/all_ops.h>
+#include <tests/cpp/utils.h>
+#include <tests/cpp/validator.h>
+#include <host_ir/lower_to_llvm.h>
+#include <multidevice/utils.h>
+
+namespace nvfuser {
+
+namespace hir {
+
+using testing::Contains;
+using HostIrLLVMTest = NVFuserTest;
+
+using ::testing::ElementsAre;
+using ::testing::HasSubstr;
+using ::testing::ThrowsMessage;
+
+// Print tensor info
+void print_tensor_info(const at::Tensor& t){
+  std::cout << "Tensor dtype: " << t.dtype() << "\n";
+  std::cout << "Shape: " << t.sizes() << "\n"; 
+  std::cout << "Strides: " << t.strides() << "\n";
+  std::cout << "Is Contiguous: " << t.is_contiguous() << "\n";
+  std::cout << "Device: " << t.device() << "\n";
+  std::cout << "Data ptr: " << t.data_ptr() << "\n";
+}
+
+void print_iter_domain(const std::vector<IterDomain*>& iter_domain, const std::string& name){
+  std::cout << name << ": ";
+  for(auto* id : iter_domain){
+    std::cout << id->toString() << " ";
+  }
+  std::cout << std::endl;
+}
+
+TEST_F(HostIrLLVMTest, AllocationMergeSplit1) {
+  Fusion fusion;
```

Comment:
This looks wrong. HostIrLlvmJit, similar to HostIrEvaluator, should take HostIrContainer, not Fusion.